src/material-experimental/mdc-list/action-list.ts

```diff
@@ -23,4 +23,11 @@ import {MatInteractiveListBase, MatListBase} from './list-base';
     {provide: MatListBase, useExisting: MatActionList},
   ]
 })
-export class MatActionList extends MatInteractiveListBase {}
+export class MatActionList extends MatListBase {
+  // An navigation list is considered interactive, but does not extend the interactive list
```

Comment:
Is this comment right? it mentions a navigation list

Reply:
@mmalerba Oh, good catch. This is a mistake that happend when I addressed feedback.

I was not aware of this discussion and I assumed it was done by accident that way. The reason I changed it over is that it seemed more accessible and matching with the existing list implementation to have _every item_ tabbable.

The way it was set up before, it didn't have any list-related aria-role, and screenreaders weren't even suggesting that there are more items that can be iterated through. Hence for compatibility and better a11y I didn't make it a full interactive list. If you think we should still do that; then that is as easy as extending from the base class (I kept that).

We should then also consider adding a role to the nav and action list, right? Though that is contradictory to what we mention in the list overview. i.e. that we do not apply any special accessibility treatment to lists (except selection-list). Let me know what you prefer here. Happy to make the change.

Reply:
I hadn't gotten to polishing the accessibility yet, but I did test the MDC demos with chromevox at least and it seemed to work well. I think if you have a list with many items having the list be a single tab stop is definitely the way to go. Otherwise the user might have to press tab quite a few times to escape the list.

I think `role=nav` makes sense for the navlist at least, not actionlist though. Unless @jelbourn had a specific reason for not doing that? I can't remember, but like I said , I didn't finish polishing the a11y, so I probably just didn't get to adding it

Reply:
Yeah, I'd agree with the point you made for lists with lots of items. I'm unsure if `role=nav` would be suitable for the navigation list as it doesn't suggest screenreader users that they are within a list. Hence it's not clear to them that there are actually more items they could iterate through, if they'd use the arrow keys. This is something that is well-defined for a listbox but not for a `nav` as it seems. That is the primary reason why I went with keeping individual items tabbable. That seems to match best with the accessibility experience one would get when using a standard `<nav>` element with a list of anchors.

Reply:
I also tested with ChomeVox and VoiceOver. Also tested the "recommended" pattern of a checkbox selection list in MDC to see if it suitable for the selection-list. This "recommendation" seemed to have a serious accessibility issue though, that has the _same root cause_ as if we'd go with `role="nav"` for nav-list. I sent an issue for the MDC team about this: [Issue for MDC](https://github.com/material-components/material-components-web/issues/6367).

Yeah, would love some feedback from @jelbourn on this too. I'm confident that the current accessibility treatment (i.e. none for lists) is most suitable and would work best with the migration (matching the non-MDC one and fulfilling what we explain in our list overview). We only apply special accessibility treatment to the selection-list currently.

Reply:
My original intent with `mat-nav-list` was to follow the nav + anchor pattern directly, since at the time I thought that it would be the most "correct" pattern for a set of navigation links. I later discovered that this isn't true- nav + anchor works well if you have maybe 3 or 4 anchors, but for longer lists all of the tabstops are cumbersome. Now my understanding is that treating the navigation list as either a list, menu, or tree would be the preferred approach, depending on the use case (obviously tree if the list has hierarchy), all of which use arrow-key interactions to move through the items.

Reply:
@jelbourn Yeah, that sounds reasonable to me. The key part is making the `navigation-list` use a list/menu/tree role so that the keyboard interaction and context is clear to screenreader users. I'm happy to make that change, but want to say again that this would be an incompatibility change with the non-MDC navigation-list (though that could be acceptable; are we good on it from your perspective?)

Also, what are we doing for the action-list? should we do the changes here are part of this PR?

Reply:
We should do one of two things:
* If MDC implements a different pattern by default for nav or for action list, we should use that
* They don't don't (or just don't have anything), we should keep things how we had them before and track something for changing them at some point in the future.

Basically we should delegate to MDC if it's better than what we have now, but otherwise not change anything.

Reply:
MDC doesn't seem to be opinionated about this. There are no concrete examples in the readme on how they'd use the list with anchors or buttons. On their example page I found a list using anchors. That one uses a list w/ arrow keys, but wrapped it inside a `<nav>` element. That's not great as it doesn't suggest that the keyboard shortcuts work, or that there are more items. 

I also found another case of a navigation list in their docs app (not an actual example). They just had individual anchor list-items wrapped in an `ul`. TL;DR: They don't seem opinionated and don't have a great solution to this; their example is lacking good accessibility as it seems.

The PR currently implements your (2) bullet point, so I think we should be good for now. I can create a JIRA for revisiting a11y for nav and action lists (i.e. due to the scenario mentioned with a lot of list items being cumbersome)

Reply:
sgtm

Reply:
@devversion and I discussed a bit offline. I think there were a few cases I didn't consider before. I agree that its best to go with this approach for now